### ISSUE TITLE
Layout constraint problem, Xcode 11 and iOS 13

### ISSUE THREAD

**Issue by JeffersonSchuler** [hackiftekhar/IQKeyboardManager#1616]
**Describe the bug**
Getting a layout constraint error now that I have upgraded to Xcode 11 and iOS 13 on a real device. Selecting a UITextfield that has "secure text entry" enabled causes the following error:

[LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x281b94b90 'accessoryView.bottom' _UIRemoteKeyboardPlaceholderView:0x104ffa200.bottom == _UIKBCompatInputView:0x108707890.top   (active)>",
    "<NSLayoutConstraint:0x281b7f840 'assistantHeight' TUISystemInputAssistantView:0x108531590.height == 45   (active)>",
    "<NSLayoutConstraint:0x281b919a0 'assistantView.bottom' TUISystemInputAssistantView:0x108531590.bottom == _UIKBCompatInputView:0x108707890.top   (active)>",
    "<NSLayoutConstraint:0x281b91680 'assistantView.top' V:[_UIRemoteKeyboardPlaceholderView:0x104ffa200]-(0)-[TUISystemInputAssistantView:0x108531590]   (active)>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x281b91680 'assistantView.top' V:[_UIRemoteKeyboardPlaceholderView:0x104ffa200]-(0)-[TUISystemInputAssistantView:0x108531590]   (active)>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKitCore/UIView.h> may also be helpful.

**Expected behavior**
No constraint issues.

### Versions
**Xcode:** Version 11.0 (11A420a)
**Mac OS:** 10.15 Beta (19A573a)
**Simulator/Device:** iOS 13.1
**Simulator/Device Name:** iPhone X
**Library Version:** 6.5.0

**Comment by RiberWang** [comment]
same as to you!

**Comment by RiberWang** [comment]
same!

**Comment by gask** [comment]
Same. And I'm worried this is happening on devices as well.

**Comment by MoElnaggar14** [comment]
I have the same problem in real devices

**Comment by GyanHajela** [comment]
I have the same problem in all simulator.

**Comment by singleD** [comment]
I think  it's a problem caused by system keyboard, because I have the same problem without this pod

**Comment by MoElnaggar14** [comment]
I thought this problem because of toolbar because when I disabled it , this problems was gone :)

**Comment by hackiftekhar** [comment]
I can confirm that the problem is related to the toolbar.

**Comment by hackiftekhar** [comment]
I had more information on this. I created a clean project with two textField one is having its inputView as a UIDatePicker and another one is a simple textField with autocorrection = default. This project does not contain our library.

I found out that this constraint violation issue occurs when we switch between those 2 textFields. I believe that this is not a bug in our library but apple's implementation of the keyboard. I'm also attaching the demo project I created.
[TextField Constraint violation Demo.zip](https://github.com/hackiftekhar/IQKeyboardManager/files/3750762/TextField.Demo.zip)

**Comment by MeiYuLong** [comment]
how to fix ,samebody done ?

**Comment by hackiftekhar** [comment]
It's a bug in Apple implementation @MeiYuLong.

**Comment by john-gaz** [comment]
Anyone found a work around?

**Comment by anelad** [comment]
Not yet I suppose.

**Comment by emreond** [comment]
I'm also having the same issue :(

**Comment by YourDaddy2000** [comment]
FIX: disable the Correction for that TextField

**Comment by lolgear** [comment]
@hackiftekhar do you know id this bug in Apple bug report?

**Comment by nilhy** [comment]
I have the same problem in all simulator. +1

**Comment by gokulprathin8** [comment]
Same issue. any fix? +1

**Comment by lolgear** [comment]
@gokulprathin8 
It isn't a problem of IQKeyboardManager, it is Apple internal bug.

**Comment by horseshoe7** [comment]
I'm not using this framework, but I am experiencing this bug.  Does anyone have a solution?  For me it's happening because I have an .inputView and an .inputAccessoryView on a UITextField.

**Comment by hackiftekhar** [comment]
> @hackiftekhar do you know id this bug in Apple bug report?

 Nope.

**Comment by hackiftekhar** [comment]
> I'm not using this framework, but I am experiencing this bug. Does anyone have a solution? For me it's happening because I have an .inputView and an .inputAccessoryView on a UITextField.

Please read previous comments for workarounds.

**Comment by SDPrio** [comment]
I am not using IQKeyboardManager but I came across the exact same problem while work with two `UITextFields` where one uses the system keyboard and one has its own, custom `inputView` (a `UIDatePicker`). When switching from the DatePicker TextField to the normal one the constraint error is shown. In rare cases it happens that the DatePicker inputView is not shown anymore when switching back.

I digged a little deeper and found out, that the custom `inputView` is places within a `UIInputSetHostView` which also contains (beside others) a `TUISystemInputAssistantView`. When the custom `inputView` is shown everything is fine and there are no conflict. However, when switching to the TextField with the standard keyboard, the height of the `TUISystemInputAssistantView` is set to 45 which conflicts with the top constraint which is set to the custom `inputView`...

**Long story short:** A hacky workaround for this problem is to manipulate the the constraints which are set `inputView` and give them a slightly lower priority :

```
func updateConstraintsWhenCustomKeyboardIsShown(customInputView: UIView) {
        if let superview = customInputView.superview {
            for constraint in superview.constraints {
                
                if (constraint.secondItem === customInputView && constraint.secondAttribute == .top) {
                    constraint.priority = UILayoutPriority(999)
                }
            }
        }
}
```

**Comment by kiviabrito** [comment]
> FIX: disable the Correction for that TextField

This fixed my problem, tks!!

**Comment by netgfx** [comment]
Any solutions for this? I already have `autocorrectionType = false` but when autofill password is enabled. But the view still jitters when I switch between 2 input fields.

**Comment by kiviabrito** [comment]
> Any solutions for this? I already have `autocorrectionType = false` but when autofill password is enabled. But the view still jitters when I switch between 2 input fields.

Did you try setting to .no instead of false?

**Comment by netgfx** [comment]
Yes I meant `.no` still nothing... I believe if the Auto-Fill password keeps appearing then the problem will still be present.

**Comment by Demircivi** [comment]
Disabling autocorrection didn't work quite well for me, my issue is autofill actually. I'm experiencing this constraint issue in my OTP screen, when my device receives the OTP SMS autofill is being shown and it causes this bug to occur.

**Comment by fabricetiennette** [comment]
In MY case i fix this error by setting textfield correction to default
<img width="255" alt="Screenshot 2020-05-05 at 11 53 06" src="https://user-images.githubusercontent.com/19342638/81054431-1fb7d980-8ec7-11ea-80ad-9ed169f9c8e0.png">

**Comment by netgfx** [comment]
This does remove the auto correction, but it doesn't remove the password/email suggestion that also takes up space above the keyboard and causes the layout issue and up/down jumping of the form.

**Comment by jimbrayrcp** [comment]
**The problem with my code was this...**

I found that I had set the toolbar this way:
`let toolBar = UIToolbar(frame: CGRect(x: 0, y: 0, width: textView.frame.size.width, height: 44.0))`
and the error was popping every time. 

**Now I have it set like this**
`let toolBar = UIToolbar(frame: CGRect(origin: .zero, size: CGSize(width: 100, height: 44.0)))`
and the error is resolved.

Hope this helps.

**Comment by ArikElb** [comment]
I am also not currently using IQKeyboardMananger but encountered this issue and based on the info here and playing around with options in Xcode, I found that assigning a Content Type in Text Input Traits (other than the default 'Unspecified') resolved the issue for me.
eg
![image](https://user-images.githubusercontent.com/19838089/84834446-45d5ab00-b029-11ea-8169-af731658815d.png)

**Comment by bugrym** [comment]
> FIX: disable the Correction for that TextField

Jun 2020, ur solution still works! Thanks a lot 🙏🏻

**Comment by netgfx** [comment]
All these solutions work for auto suggestions etc, however the issue is still present on password/email suggestions that cannot be disabled unfortunately, so the "jumping" of the form still exists.

**Comment by aksthelion** [comment]
Following solution doesn't work for me.

> **The problem with my code was this...**
> 
> I found that I had set the toolbar this way:
> `let toolBar = UIToolbar(frame: CGRect(x: 0, y: 0, width: textView.frame.size.width, height: 44.0))`
> and the error was popping every time.
> 
> **Now I have it set like this**
> `let toolBar = UIToolbar(frame: CGRect(origin: .zero, size: CGSize(width: 100, height: 44.0)))`
> and the error is resolved.
> 
> Hope this helps.

**Comment by xingheng** [comment]
I used `UITextView` and found this, the auto-correction and spell checking don't work for me, either.

```
textView = UITextView()
textView.autocapitalizationType = .none
textView.autocorrectionType = .no
textView.spellCheckingType = .no
textView.smartQuotesType = .no
textView.smartDashesType = .no
textView.smartInsertDeleteType = .no
```

**Comment by lexuanquynh** [comment]
it't not working with. me

**Comment by wzj583585700** [comment]
苹果这么垃圾吗

**Comment by bobbelkl** [comment]
Just use a non-default keyboard type if you are using a UIDatePicker for the accessoryView.
e.g. keyboardType = .phonePad

**Comment by AlxndrIvnch** [comment]
In case I want to show suggestions for some field, like - textField.textContentType = .telephoneNumber, then I have to disable IQKeyboardManager toolbar for that VC or ignore constraints warning, no other solution?

**Comment by zafarivaev** [comment]
Setting `textField.textContentType = .oneTimeCode` removed the Auto Layout warnings for me
